client/data/plugins/installed/use-plugins-query.js

```diff
@@ -0,0 +1,43 @@
+import { useQuery } from 'react-query';
+import { useDispatch } from 'react-redux';
+import wpcom from 'calypso/lib/wp';
+import { receiveSitePlugins, updatePlugin } from 'calypso/state/plugins/installed/actions';
+
+export function getCacheKey( siteIds ) {
+	if ( siteIds.length === 1 ) {
+		return [ 'plugins', siteIds[ 0 ] ];
+	}
+	// The `dev` and `view` query params are not included in the cache key because we want all
+	// the hooks to have the same idea of what the current view is, regardless of dev flags.
+	return [ 'all-plugins' ];
+}
+
+const usePluginsQuery = ( siteIds, queryOptions = {} ) => {
+	let fetch = () => wpcom.req.get( '/me/sites/plugins' );
+	const siteId = siteIds[ 0 ];
+
+	if ( siteIds.length === 1 ) {
+		fetch = () => wpcom.site( siteId ).pluginsList();
+	}
+
+	const dispatch = useDispatch();
+	return useQuery( getCacheKey( siteIds ), fetch, {
+		...queryOptions,
+		onSuccess: ( data ) => {
+			if ( data.sites ) {
+				for ( const [ siteIdKey, plugins ] of Object.entries( data.sites ) ) {
+					dispatch( receiveSitePlugins( siteIdKey, plugins ) );
```

Comment:
Is there a reason why we continue using Redux for storing and later accessing the installed plugins? That way `react-query`'s cache is not optimally leveraged. Have you planned to use plugins that are the result of the `react-query` query?

Also, do we have a plan for keeping the `react-query` and Redux plugins in sync? Right now it seems like they can easily get out of sync if we use the Redux version to fetch the plugins (like we do in many areas).

Reply:
Currently, we don't use the `react-query` response for plugin data. We only use it for knowing if the plugin was isFetching to display the placeholder UI. The reason why is mainly because I wanted to keep the refactor of this code small. I think to try to switch over to only use `react-query` we would also need to update all the mutation calls. ( update, install, activate, enable autoupdate and remove plugin calls. Which I wanted to avoid for now.

Reply:
I see 👍  It makes sense to separate the actual plugins data from the requesting state. However, as pointed above, we'll still need to make sure we trigger the `react-query` query everytime we request plugins with Redux, in order to maintain consistency.